cmd/helm/search_repo.go

```diff
@@ -154,16 +154,27 @@ func (o *searchRepoOptions) applyConstraint(res []*search.Result) ([]*search.Res
 
 	data := res[:0]
 	foundNames := map[string]bool{}
+	appendSearchResults := func(res *search.Result) {
+		data = append(data, res)
+		if !o.versions {
+			foundNames[res.Name] = true // If user hasn't requested all versions, only show the latest that matches
+		}
+	}
 	for _, r := range res {
 		if _, found := foundNames[r.Name]; found {
 			continue
 		}
 		v, err := semver.NewVersion(r.Chart.Version)
-		if err != nil || constraint.Check(v) {
-			data = append(data, r)
-			if !o.versions {
-				foundNames[r.Name] = true // If user hasn't requested all versions, only show the latest that matches
+
+		if err != nil {
+			// if pre-release start with zero and not devel mode, ingore
+			// This is to fix the bug that the version number of start with zero was retrieved in the helm search repo
+			if err == semver.ErrSegmentStartsZero && !o.devel {
```

Comment:
This will handle the case of the leading `0` on a prerelease. There is a second possible prerelease error that could be returned of `semver.ErrInvalidPrerelease`. You can find the code that checks this at https://github.com/Masterminds/semver/blob/60c7ae8a99210a90a9457d5de5f6dcbc4dab8e64/version.go#L575-L592.

Could you please update the code to handle both of these possible prerelease error codes.

Reply:
done